L1Trigger/L1THGCal/plugins/geometries/HGCalTriggerGeometryV9Imp1.cc

```diff
@@ -0,0 +1,1100 @@
+                        if(wafer_itr==wafers_old_to_wafers_.end()) invalid_triggercells_.emplace(neighbor);
+                        else if(!validTriggerCellFromCells(neighbor)) invalid_triggercells_.emplace(neighbor);
+                    }
+                }
+            }
+            for(int trigger_cell=0; trigger_cell<number_trigger_cells_in_wafers_.at(waferfh_type); trigger_cell++)
+            {
+                std::set<unsigned> trigger_cell_ids;
+                trigger_cell_ids.emplace(HGCalDetId(ForwardSubdetector::HGCHEF, -1, layer, 1, trigger_wafer, trigger_cell));
+                trigger_cell_ids.emplace(HGCalDetId(ForwardSubdetector::HGCHEF, 1, layer, 1, trigger_wafer, trigger_cell));
+                for(unsigned trigger_cell : trigger_cell_ids)
+                {
+                    if(!validTriggerCellFromCells(trigger_cell)) invalid_triggercells_.emplace(trigger_cell);
+                    for(unsigned neighbor : getNeighborsFromTriggerCell(trigger_cell))
+                    {
+                        auto wafer_itr = wafers_old_to_wafers_.find(HGCalDetId(neighbor).wafer());
+                        if(wafer_itr==wafers_old_to_wafers_.end()) invalid_triggercells_.emplace(neighbor);
+                        else if(!validTriggerCellFromCells(neighbor)) invalid_triggercells_.emplace(neighbor);
+                    }
+                }
+            }
+        }
+    }
+    unsigned n_layers_hsc = hscTopology().dddConstants().layers(true);
+    unsigned first_layer_hsc = hscTopology().dddConstants().firstLayer();
+    for(unsigned layer=first_layer_hsc; layer<=first_layer_hsc+n_layers_hsc; layer++)
+    {
+        int type =  hscTopology().dddConstants().getTypeTrap(layer);
+        for(const auto& module_ncells : number_trigger_cells_in_wafers_sci_)
+        {
+            unsigned trigger_wafer = module_ncells.first;
+            // loop on the trigger cells in each wafer
+            for(int trigger_cell=1; trigger_cell<module_ncells.second; trigger_cell++)
+            {
+                std::set<unsigned> trigger_cell_ids;
+                trigger_cell_ids.emplace(HGCalDetId(ForwardSubdetector::HGCHEB, -1, layer, type, trigger_wafer, trigger_cell));
+                trigger_cell_ids.emplace(HGCalDetId(ForwardSubdetector::HGCHEB, 1, layer, type, trigger_wafer, trigger_cell));
+                for(unsigned trigger_cell : trigger_cell_ids)
+                {
+                    if(!validTriggerCellFromCells(trigger_cell)) invalid_triggercells_.emplace(trigger_cell);
+                    for(unsigned neighbor : getNeighborsFromTriggerCell(trigger_cell))
+                    {
+                        if(!validTriggerCellFromCells(neighbor)) invalid_triggercells_.emplace(neighbor);
+                    }
+                }
+            }
+        }
+    }
+}
+
+unsigned 
+HGCalTriggerGeometryV9Imp1::
+packWaferCellId(unsigned subdet, unsigned wafer, unsigned cell) const
```

Comment:
it seems suboptimal to copy all the packing/unpacking code from the DetIds here, but I can't immediately think of a better way to get specific pieces of the bit string.

Reply:
An other possibility could be to pack the full detid, with all the other values set to 0.

Reply:
I kept the same `pack...` functions for the moment